### PR TITLE
Simplify journal writer response

### PR DESCRIPTION
**Pull request by bf8086**
Ratis by default echo the journal message back after it is applied. This wastes a lot of memory and CPU time. This change makes the state machine explicitly return empty message since the journal write doesn't really care about the response content.

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java [hunk 1]
```diff
@@ -242,7 +246,10 @@ public void close() {
   public CompletableFuture<Message> applyTransaction(TransactionContext trx) {
     try {
       applyJournalEntryCommand(trx);
-      return super.applyTransaction(trx);
+      RaftProtos.LogEntryProto entry = Objects.requireNonNull(trx.getLogEntry());
+      updateLastAppliedTermIndex(entry.getTerm(), entry.getIndex());
+      // no response message is expected
```
Comment (gpang) [inline core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java:251]: Can you add more context/details to this comment?

Also, is there a way to add a simple unit test for this behavior?

**Review by gpang — CHANGES_REQUESTED** [review]
@bf8086 Thanks!

File: core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java (hunk 1 above)
Comment (bf8086) [reply core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java]: Added a simple test and more comments.

**Comment by gpang** [comment]
Not sure what this means?
```
Error: 3.115 [ERROR] alluxio.client.fs.BufferedBlockInStreamIntegrationTest  Time elapsed: 0.653 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at alluxio.master.AbstractLocalAlluxioCluster.setAlluxioWorkDirectory(AbstractLocalAlluxioCluster.java:363)
	at alluxio.master.LocalAlluxioCluster.initConfiguration(LocalAlluxioCluster.java:125)
	at alluxio.testutils.LocalAlluxioClusterResource.start(LocalAlluxioClusterResource.java:147)
	at alluxio.testutils.LocalAlluxioClusterResource$1.evaluate(LocalAlluxioClusterResource.java:182)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.RuntimeException: Failed to create testing directory /tmp/alluxio-tests
	at alluxio.AlluxioTestDirectory.createTestingDirectory(AlluxioTestDirectory.java:80)
	at alluxio.AlluxioTestDirectory.<clinit>(AlluxioTestDirectory.java:38)
	... 15 more
```

**Comment by bf8086** [comment]
Test works locally, will rerun the actions.

**Review by gpang — APPROVED** [review]
LGTM

**Comment by bf8086** [comment]
alluxio-bot, merge this please.

**Comment by bf8086** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please.

**Pull request merged**